DQM/HGCal/plugins/HGCalTriggerClient.cc

```diff
@@ -84,103 +204,434 @@ void HGCalTriggerClient::analyze(const edm::Event& iEvent, const edm::EventSetup
+  for (unsigned i(cafe_word_loc[1] + 1); i < daq_event_size[1] + unsigned(cafe_word_loc[1]) + 1; i++) {
+    const uint32_t wordL = (payload[i] & 0xFFFFFFFF);
+    const uint32_t wordM = ((payload[i] >> 32) & 0xFFFFFFFF);
 
-  // //map also the cell types
-  // const auto& siCellInfo = iSetup.getData(siModuleInfoToken_);
-  // eleidtoSiInfo_ = hgcal::mapEleIdToSiInfo(moduleInfo, siCellInfo);
+    const uint32_t energy1 = wordL & 0x7F;
+    const uint32_t location1 = (wordL >> 7) & 0x3F;
+    const uint32_t energy2 = (wordL >> 13) & 0x7F;
+    const uint32_t location2 = (wordL >> (13 + 7)) & 0x3F;
+    const uint32_t bx_counter = (wordL >> 26) & 0xF;
 
+    const uint32_t energy3 = wordM & 0x7F;
+    const uint32_t location3 = (wordM >> 7) & 0x3F;
+    const uint32_t energy4 = (wordM >> 13) & 0x7F;
+    const uint32_t location4 = (wordM >> (13 + 7)) & 0x3F;
+    const uint32_t bx_counter1 = (wordM >> 26) & 0xF;
+
+    energy_unpkd[0][index_ibx][index_stc] = energy1;
+    energy_unpkd[0][index_ibx][index_stc + 6] = energy2;
+    energy_unpkd[1][index_ibx][index_stc] = energy3;
+    energy_unpkd[1][index_ibx][index_stc + 6] = energy4;
+
+    loc_unpkd[0][index_ibx][index_stc] = location1;
+    loc_unpkd[0][index_ibx][index_stc + 6] = location2;
+    loc_unpkd[1][index_ibx][index_stc] = location3;
+    loc_unpkd[1][index_ibx][index_stc + 6] = location4;
+
+    bx_unpkd[0][index_ibx][index_stc] = bx_counter;
+    bx_unpkd[0][index_ibx][index_stc + 6] = bx_counter;
+    bx_unpkd[1][index_ibx][index_stc] = bx_counter1;
+    bx_unpkd[1][index_ibx][index_stc + 6] = bx_counter1;
+
+    if (verbose_ >= 3) {
+      out << "Event : " << iEvent.id().event() << "LSB :  STC1 " << index_stc << ", STC2 " << index_stc + 6 << ", ibx "
+          << index_ibx << ", EnergyL1 : " << energy_unpkd[0][index_ibx][index_stc]
+          << ", locL1 : " << (loc_unpkd[0][index_ibx][index_stc] & 0x3)
+          << ", bx_counterL1 : " << bx_unpkd[0][index_ibx][index_stc] << "\n"
+          << "\t\tEnergyL2 : " << energy_unpkd[0][index_ibx][index_stc + 6]
+          << ", locL2 : " << (loc_unpkd[0][index_ibx][index_stc + 6] & 0x3)
+          << ", bx_counterL2 : " << bx_unpkd[0][index_ibx][index_stc + 6] << "\n";
+      out << "\t MSB :  STC3 " << index_stc << ", STC4 " << index_stc + 6 << ", ibx " << index_ibx
+          << ", EnergyL3 : " << energy_unpkd[1][index_ibx][index_stc]
+          << ", locL3 : " << (loc_unpkd[1][index_ibx][index_stc] & 0x3)
+          << ", bx_counterL3 : " << bx_unpkd[1][index_ibx][index_stc] << "\n"
+          << "\t\tEnergyL4 : " << energy_unpkd[1][index_ibx][index_stc + 6]
+          << ", locL4 : " << (loc_unpkd[1][index_ibx][index_stc + 6] & 0x3)
+          << ", bx_counterL4 : " << bx_unpkd[1][index_ibx][index_stc + 6] << "\n";
+    }
+
+    if (index_stc == 5)
+      index_stc = 0;
+    else
+      index_stc++;
+    if (index_stc == 0)
+      index_ibx++;
+  }
+
+  //check bx
+  for (int iect = 0; iect < 2; iect++) {
```

Comment:
This is the first of a series of triple-nested `for` loops. Does it make sense to to the same loop multiple times or could all checks be done in one (triple-nested) loop?

Reply:
Necessary modifications have been applied (see [8165d07](https://github.com/CMS-HGCAL/cmssw/pull/86/commits/8165d077af54e5ab2fcceb6c538b92669bbe113d))